File: typescript/Creational/01-FactoryMethod/Concrete Products/CarpentryJobCard.ts
class CarpentryJobCard implements JobCard {
  readonly jobId    = shortId();
  readonly category = "Carpentry";

  constructor(
    readonly title:    string,
    readonly priority: string
  ) {}

  printSummary(): void {
    console.log(`[${this.category}] #${this.jobId} — ${this.title} (Priority: ${this.priority})`);
    console.log("  → Requires: Carpenter + material estimate");
  }
}

File: csharp/Creational/01-FactoryMethod/Concrete Products/CarpentryJobCard .cs
public class CarpentryJobCard : IJobCard
{
    public string JobId    { get; } = Guid.NewGuid().ToString()[..8];
    public string Title    { get; }
    public string Category => "Carpentry";
    public string Priority { get; }

    public CarpentryJobCard(string title, string priority)
    {
        Title    = title;
        Priority = priority;
    }

    public void PrintSummary() =>
        Console.WriteLine(
            $"[{Category}] #{JobId} — {Title} (Priority: {Priority})" +
            $"\n  → Requires: Carpenter + material estimate");
}